src/app/containers/RadioSchedule/index.jsx

```diff
@@ -9,14 +13,76 @@ const logger = webLogger();
 
 const RadioScheduleContainer = ({ endpoint }) => {
   const { enabled } = useToggle('radioSchedule');
-  const { hasRadioSchedule } = useContext(ServiceContext);
+  const {
+    hasRadioSchedule,
+    service,
+    script,
+    dir,
+    timezone,
+    locale,
+  } = useContext(ServiceContext);
   const radioScheduleEnabled = enabled && hasRadioSchedule;
 
   const [schedule, setRadioSchedule] = useState([]);
 
+  const getProgramState = (currentTime, startTime, endTime) => {
+    // live - currentTime < endTime && currentTime > startTime
+    if (currentTime < endTime && currentTime > startTime) {
+      return 'live';
+    }
+    // onDemand - is live that just ended
+    if (currentTime > endTime) {
+      return 'onDemand';
+    }
+    // next - is immedeately after what's live
+    return 'next';
+  };
+
+  const getLink = (state, latestProgram) => {
+    const url = `/${service}/${latestProgram.serviceId}`;
+    return state === 'live'
+      ? `${url}/liveradio`
+      : `${url}/${latestProgram.broadcast.pid}`;
+  };
+
   const handleResponse = async response => {
     const radioScheduleData = await response.json();
-    setRadioSchedule(radioScheduleData.schedules);
+
+    const currentTime = parseInt(moment.utc().format('x'), 10);
+    // finding latest programme, that may or may not still be live. this is because there isn't
+    // always a live programme, in which case we show the most recently played programme on demand.
+    const latestProgrammeIndex = findLastIndex(
+      propSatisfies(time => time < currentTime, 'publishedTimeStart'),
+    )(radioScheduleData.schedules);
+
+    const radioSchedules = radioScheduleData.schedules;
+    const schedulesToShow = [
+      radioSchedules[latestProgrammeIndex],
+      radioSchedules[latestProgrammeIndex - 1],
+      radioSchedules[latestProgrammeIndex - 2],
+      radioSchedules[latestProgrammeIndex + 1],
```

Comment:
I'm thinking of scenarios where `latestProgrammeIndex` is either `0` or is the last item in the `radioschedules` array, that would cause `latestProgrammeIndex - 1` or `latestProgrammeIndex + 1` to be undefined. We should have some error handling for these scenarios unless we're absolutely positive they can never occur.

Reply:
I think we should check that all of these are defined, and if not, we shouldn't try and render the component :)

Reply:
This is now being checked. Currently if we don't have any programs in the data or have incomplete data (e.g. just 2 programs), the schedule won't get rendered and there won't be any console errors.

Reply:
Also, I am only checking if `radioSchedules[latestProgrammeIndex - 2]`  and `radioSchedules[latestProgrammeIndex + 1]` are defined, since if they are, `radioSchedules[latestProgrammeIndex - 1]` and `radioSchedules[latestProgrammeIndex]` will be defined by default

Reply:
Will also add a unit test that checks that a schedule isn't rendered if data is incomplete